docs/import/upload_image_collection.js

```diff
@@ -178,35 +164,41 @@ function uploadFileToGCS(name, contents, collectionName, callback) {
     headers: gcsHeader,
     body: contents,
   };
-  fetch(URL, request)
-      .then((r) => r.json())
-      .then((r) => {
-        uploadedToGCS++;
-        return r;
-      })
-      .catch((err) => {
-        console.error(err);
-        resultDiv.innerHTML += '<br>Error uploading ' + name + ': ' + err;
-        throw err;
-      })
-      .then((result) => callback(result.bucket, collectionName, name))
-      .catch((err) => {
-        console.error(err);
-        resultDiv.innerHTML += '<br>Error processing ' + name + ': ' + err;
-      });
+  let bucket;
+  try {
+    const result = await fetch(URL, request);
+    const json = await result.json();
+    if (result.status !== 200) {
+      throw new Error('Unable to fetch from GCS: ' + json.error.message);
+    }
+    ({bucket} = json);
+  } catch (err) {
+    resultDiv.innerHTML += '<br>Error uploading ' + name + ': ' + err;
+    showError(err, 'Error uploading ' + name);
+    throw err;
+  }
+  uploadedToGCS++;
+  try {
+    return callback(bucket, qualifiedName, name);
+  } catch (err) {
+    resultDiv.innerHTML += '<br>Error processing ' + name + ': ' + err;
+    showError(err, 'Error processing ' + name);
+    throw err;
+  }
 }
 
 /**
- * Checks to see if asset given by collectionName is present. If not, creates it
- * and makes it world-readable.
+ * Checks to see if asset given by `qualifiedName` is present. If not, creates
+ * it and makes it world-readable.
  *
- * @param {string} collectionName Must contain only legal EE characters
- * @return {Promise<undefined>} Promise to wait for operation completion on
+ * @param {string} qualifiedName Must contain only legal EE characters. Like
+ *     '2017-harvey/noaa-images'
+ * @return {Promise<void>} Promise to wait for operation completion on
```

Comment:
I know this isn't new, but having a difficult time parsing this return description

Reply:
Rewritten. Better?